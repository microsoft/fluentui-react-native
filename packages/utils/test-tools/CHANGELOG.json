{
  "name": "@fluentui-react-native/test-tools",
  "entries": [
    {
      "date": "Thu, 26 Aug 2021 04:48:40 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.3",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "b8c866644b2c5b64530990e59153424b0d6f5dcc",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Wed, 18 Aug 2021 23:00:57 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.2",
            "author": "ruaraki@microsoft.com",
            "commit": "345d1de930ff42359260da7f1e68a16059f4d763",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Fri, 13 Aug 2021 16:12:26 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.1",
            "author": "t-lindaweng@microsoft.com",
            "commit": "8946966df6ddc0ef1c0b43e576a4124c678d7c89",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Thu, 12 Aug 2021 17:23:44 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.0",
            "author": "ruaraki@microsoft.com",
            "commit": "e524662e3e40fe45a2dcc6e2acd3aca1197b5b5a",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 10 Aug 2021 21:12:30 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.10.0",
            "author": "ruaraki@microsoft.com",
            "commit": "ce49f33c08add7892191cf840d2fdcccfd0a16bb",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Sat, 07 Aug 2021 00:40:04 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "jasonmo@microsoft.com",
            "commit": "e506337e109ef24e675c2ac794179db86fcb7e86",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Fri, 06 Aug 2021 00:36:02 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "ruaraki@microsoft.com",
            "commit": "56ccce452d0dbbd5c24f53ebdd467f2d01dda694",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Wed, 04 Aug 2021 06:26:25 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "68dd61bd45ca8ffacc1580e468f3c01810e3093e",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 27 Jul 2021 22:17:20 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.2",
            "author": "jasonmo@microsoft.com",
            "commit": "1cd8611e1e88848ab66f7c0e9113c946f968260c",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 27 Jul 2021 18:55:58 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.1",
            "author": "4123478+tido64@users.noreply.github.com",
            "commit": "07d04d166b13ff79e07a8c4cbc2e06afb1e4c3e6",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Mon, 26 Jul 2021 20:41:04 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.1",
            "author": "jasonmo@microsoft.com",
            "commit": "ad02fbbe9eaa5fbe6ec6126ccb719a1280f108a6",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    }
  ]
}
