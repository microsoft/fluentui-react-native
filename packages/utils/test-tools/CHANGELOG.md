# Change Log - @fluentui-react-native/test-tools

This log was last generated on Thu, 26 Aug 2021 04:48:40 GMT and should not be manually modified.

<!-- Start content -->

## 0.1.1

Thu, 26 Aug 2021 04:48:40 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.3 (67026167+chiuam@users.noreply.github.com)

## 0.1.1

Wed, 18 Aug 2021 23:00:57 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.2 (ruaraki@microsoft.com)

## 0.1.1

Fri, 13 Aug 2021 16:12:26 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.1 (t-lindaweng@microsoft.com)

## 0.1.1

Thu, 12 Aug 2021 17:23:44 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.0 (ruaraki@microsoft.com)

## 0.1.1

Tue, 10 Aug 2021 21:12:30 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.10.0 (ruaraki@microsoft.com)

## 0.1.1

Wed, 04 Aug 2021 06:26:25 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.9.3 (67026167+chiuam@users.noreply.github.com)

## 0.1.1

Tue, 27 Jul 2021 22:17:20 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.9.2 (jasonmo@microsoft.com)

## 0.1.1

Mon, 26 Jul 2021 20:41:04 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.9.1 (jasonmo@microsoft.com)
