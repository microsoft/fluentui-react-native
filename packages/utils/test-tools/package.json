{
  "name": "@fluentui-react-native/test-tools",
  "version": "0.1.1",
  "description": "Tools and mocks for testing components using jest and enzyme",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "private": true,
  "scripts": {
    "build": "fluentui-scripts build",
    "just": "fluentui-scripts",
    "clean": "fluentui-scripts clean",
    "depcheck": "fluentui-scripts depcheck",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "verify-api": "fluentui-scripts verify-api-extractor",
    "update-api": "fluentui-scripts update-api-extractor"
  },
  "dependencies": {
    "@fluentui-react-native/theme-types": ">=0.3.0 <1.0.0"
  },
  "devDependencies": {
    "@types/react": "^16.9.34",
    "@types/react-native": "^0.62.0",
    "@uifabricshared/build-native": "^0.1.1",
    "@uifabricshared/eslint-config-rules": "^0.1.1",
    "react": "16.11.0",
    "react-native": "0.62.2"
  },
  "peerDependencies": {
    "react": "^16.8.0",
    "react-native": ">=0.60.0"
  },
  "author": "",
  "license": "MIT"
}
