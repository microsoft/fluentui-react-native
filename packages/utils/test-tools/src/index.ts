export * from './enzymeTests';
export * from './mockTheme';
