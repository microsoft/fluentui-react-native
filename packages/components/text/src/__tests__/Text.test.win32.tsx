import * as React from 'react';
import { Text } from '..';
import * as renderer from 'react-test-renderer';
import { checkReRender, checkRenderConsistency } from '@fluentui-react-native/test-tools';

it('Text default', () => {
  const tree = renderer.create(<Text>Text default</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Text all props', () => {
  const tree = renderer
    .create(
      <Text disabled variant="bodyStandard">
        All props
      </Text>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Text all tokens', () => {
  const BoldText = Text.customize({
    tokens: {
      fontFamily: 'Wingdings',
      fontWeight: '900',
      fontSize: 20,
    },
  });
  const tree = renderer.create(<BoldText>All tokens</BoldText>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Text simple rendering does not invalidate styling', () => {
  checkRenderConsistency(() => <Text>Default</Text>);
  checkRenderConsistency(() => <Text disabled>Default</Text>);
  checkRenderConsistency(() => <Text variant="headerStandard">Default</Text>);
});

it('Text re-renders correctly', () => {
  checkReRender(() => <Text>Default</Text>);
  checkReRender(() => <Text disabled>Default</Text>);
  checkReRender(() => <Text variant="headerStandard">Default</Text>);
});

it('Text shares produced styles across multiple renders', () => {
  const style = { color: 'black' };
  checkRenderConsistency(() => <Text style={style}>Default</Text>);
  checkRenderConsistency(() => (
    <Text style={style} disabled>
      Default
    </Text>
  ));
  checkRenderConsistency(() => (
    <Text style={style} variant="headerStandard">
      Default
    </Text>
  ));
});

it('Text re-renders correctly with style', () => {
  const style = { color: 'blue' };
  checkReRender(() => <Text style={style}>Default</Text>);
  checkReRender(() => (
    <Text style={style} disabled>
      Default
    </Text>
  ));
  checkReRender(() => (
    <Text style={style} variant="headerStandard">
      Default
    </Text>
  ));
});
