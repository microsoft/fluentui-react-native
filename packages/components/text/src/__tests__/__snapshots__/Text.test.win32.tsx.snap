// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text all props 1`] = `
<Text
  style={
    Object {
      "color": "pink",
      "fontFamily": "Segoe UI",
      "fontSize": 12,
      "fontWeight": "500",
      "margin": 0,
    }
  }
>
  All props
</Text>
`;

exports[`Text all tokens 1`] = `
<Text
  style={
    Object {
      "color": "orange",
      "fontFamily": "Wingdings",
      "fontSize": 15,
      "fontWeight": "900",
      "margin": 0,
    }
  }
>
  All tokens
</Text>
`;

exports[`Text default 1`] = `
<Text
  style={
    Object {
      "color": "#505050",
      "fontFamily": "Segoe UI",
      "fontSize": 12,
      "fontWeight": "500",
      "margin": 0,
    }
  }
>
  Text default
</Text>
`;
